### ISSUE TITLE
vscode - extension breaks the javascript debug terminal

### ISSUE THREAD

**Issue by DetachHead** [oven-sh/bun#5208]
### What version of Bun is running?

1.0.1

### What platform is your computer?

Darwin 22.1.0 x86_64 i386

### What steps can reproduce the bug?

1. install the `oven.bun-vscode` extension
2. hover over an npm script in `package.json` and click "Debug Script"
    ![image](https://github.com/oven-sh/bun/assets/57028336/4bd93c3d-fc2f-4c6d-9d94-ead28f3cf204)

### What is the expected behavior?

either:
- runs the script in the nodejs debugger with a javascript debug terminal, unaffected by bun
- runs the script in a bun debug terminal (see #5204)

### What do you see instead?

"JavaScript Debug Terminal" appears and does nothing, and "Terminal has already been disposed" error displays on the bottom right

![image](https://github.com/oven-sh/bun/assets/57028336/3f391f68-231a-4b85-a957-9f247d75350e)

**Comment by andrewbranch** [comment]
Just discovered this with slightly different repro steps and symptoms. I usually use the JavaScript Debug Terminal by manually opening it from the Terminal panel: 

<img width="361" alt="context menu for add terminal" src="https://github.com/oven-sh/bun/assets/3277153/78fa37a7-a6d2-45aa-92f4-d173bb6d6cbe">

Running any Node.js process from this terminal is supposed to auto-attach the debugger. When the extension is installed, commands from this terminal execute normally, but the debugger never attaches. (Running `bun` from this terminal also does not initiate any debug functionality.)

**Comment by pokey** [comment]
Fwiw, in addition to the aforementioned issue, I find that nothing happens when I run the Bun: Run File (`extension.bun.runFile`) or Bun: Debug File (`extension.bun.debugFile`) commands to run the currently opened script. No error, just silently does nothing. I'm guessing these have the same root cause as this issue, as it appears from the code that they both factor through the [same functions](https://github.com/oven-sh/bun/blob/d8ca6ca6995072d70e5af82bd7a19a36257b71d0/packages/bun-vscode/src/features/debug.ts#L59-L77)

Note that when I use the hover action, the JavaScript Debug Terminal does open, but nothing gets typed in there and nothing runs. If I then manually run the script by typing eg `bun myScriptName` into the JavaScript Debug Terminal, it runs and the debugger attaches properly, which did not occur before I installed the extension

**Comment by blimmer** [comment]
Just discovered this same issue and easily isolated it to the Bun VSCode plugin. To reproduce the issue:

1. Download VSCode from https://code.visualstudio.com/download
2. Set up "portable mode" to isolate the issue entirely. Setting up portable mode depends on your platform: https://code.visualstudio.com/docs/editor/portable
3. Create a test javascript file:

    ```js
    // test.js
	function main() {
	    debugger;
	}
	
	main();
    ```
4. Open a Javascript Debug Terminal and run `node test.js`. You'll see it hit the breakpoint.
5. Install the Bun VSCode extension and re-run the previous steps. The breakpoint is not longer hit.

<div>
    <a href="https://www.loom.com/share/666312ca4c2a4fbba8b99ffb40738f0b">
      <p>Bun VSCode JavaScript Debug Terminal Issue - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/666312ca4c2a4fbba8b99ffb40738f0b">
      <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/666312ca4c2a4fbba8b99ffb40738f0b-with-play.gif">
    </a>
  </div>

**Comment by zardoy** [comment]
Those who have already read the Readme.md are aware, but you can disable JavaScript Debug Terminal integration and debug Bun from the launch.json config or the F5 menu instead:

```json
{
  "bun.debugTerminal.enabled": false
}
```

Or you can toggle the setting with [Toggle](https://marketplace.visualstudio.com/items?itemName=rebornix.toggle) when you need to debug Bun (remember to create a new terminal after that).

**Comment by RiskyMH** [comment]
I'm pretty sure I fixed this in #15801

**Comment by blimmer** [comment]
Yep, confirmed that I can't reproduce this via my steps above: https://github.com/oven-sh/bun/issues/5208#issuecomment-1979794789. Thanks!
